dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/programstageworkinglistdefinition/DefaultProgramStageWorkingListDefinitionService.java

```diff
@@ -0,0 +1,257 @@
+
+    private final ProgramStageService programStageService;
+
+    private final OrganisationUnitService organisationUnitService;
+
+    private final TrackedEntityAttributeService teaService;
+
+    private final DataElementService dataElementService;
+
+    public DefaultProgramStageWorkingListDefinitionService(
+        ProgramStageWorkingListDefinitionStore programStageWorkingListDefinitionStore, ProgramService programService,
+        ProgramStageService programStageService, OrganisationUnitService organisationUnitService,
+        TrackedEntityAttributeService teaService, DataElementService dataElementService )
+    {
+        checkNotNull( programStageWorkingListDefinitionStore );
+        checkNotNull( programService );
+        checkNotNull( programStageService );
+        checkNotNull( organisationUnitService );
+        checkNotNull( teaService );
+        checkNotNull( dataElementService );
+
+        this.programStageWorkingListDefinitionStore = programStageWorkingListDefinitionStore;
+        this.programService = programService;
+        this.programStageService = programStageService;
+        this.organisationUnitService = organisationUnitService;
+        this.teaService = teaService;
+        this.dataElementService = dataElementService;
+    }
+
+    @Override
+    public long add( ProgramStageWorkingListDefinition programStageWorkingListDefinition )
+    {
+        programStageWorkingListDefinitionStore.save( programStageWorkingListDefinition );
+        return programStageWorkingListDefinition.getId();
+    }
+
+    @Override
+    public void delete( ProgramStageWorkingListDefinition programStageWorkingListDefinition )
+    {
+        programStageWorkingListDefinitionStore.delete( programStageWorkingListDefinition );
+    }
+
+    @Override
+    public void update( ProgramStageWorkingListDefinition programStageWorkingListDefinition )
+    {
+        programStageWorkingListDefinitionStore.update( programStageWorkingListDefinition );
+    }
+
+    @Override
+    public ProgramStageWorkingListDefinition get( long id )
+    {
+        return programStageWorkingListDefinitionStore.get( id );
+    }
+
+    @Override
+    public List<ProgramStageWorkingListDefinition> getAll()
+    {
+        return programStageWorkingListDefinitionStore.getAll();
+    }
+
+    @Override
+    public List<String> validate( ProgramStageWorkingListDefinition programStageWorkingListDefinition )
+    {
+        List<String> errors = new ArrayList<>();
+
+        validateProgram( errors, programStageWorkingListDefinition );
+        validateProgramStage( errors, programStageWorkingListDefinition );
+        validateName( errors, programStageWorkingListDefinition );
+
+        ProgramStageQueryCriteria queryCriteria = programStageWorkingListDefinition.getProgramStageQueryCriteria();
+
+        if ( queryCriteria == null )
+        {
+            return errors;
+        }
+
+        validateDateFilterPeriods( errors, queryCriteria );
```

Comment:
This is also outdated, but still interesting. 
It might be a better design to validate during the creation of the object, the thing is that I think that's something we should agree as a team, because currently this is validated in the bundle hook objects, this one in particular is in ProgramStageWorkingListObjectBundleHook, as it is for many other objects. That's why I think we should agree on one of the two options and stick to it.